### ISSUE TITLE
License

### ISSUE THREAD

**Issue by uliska** [wbsoft/python-ly#4]
Was the GPL deliberately chosen over the LGPL or was it simply copied over from Frescobaldi? If the latter is the case I'd ask @wbsoft to consider it and make a deliberate decision.

**Comment by wbsoft** [comment]
I did choose GPL deliberately for Frescobaldi, but I could imagine choosing the LGPL for python-ly. Is there a reason to change the license?

**Comment by uliska** [comment]
The implications are as follows:

The GPL requires anybody who writes a program using python-ly and wants 
to distribute this program to do this under the GPL also. That means 
closed-source programs cannot use the library.

The LGPL allows closed-source programs to *use* python-ly and to 
distribute their program, but of course any *changes* to python-ly would 
have to be redistributed open source too.

Or course when someone writes a program for his personal needs he can do 
so under both licenses, even for commercial purposes, it's only 
forbidden to *distribute* such a program.

I think python-ly is a good example for a library where 
https://www.gnu.org/licenses/why-not-lgpl suggests using the GPL and not 
the LGPL because python-ly offers something that can't be simply 
obtained somewhere else. So using the GPL would rule out commercial 
developers and "give free software developers a benefit".

My suggestion would be to use the LGPL and allow python-ly to be used 
even in commercial applications, but of course it is up to you. It's 
just that it should be a deliberate decision.

**Comment by crantila** [comment]
I agree it should be a deliberate decision, but unlike Urs I would encourage retaining the GPL, for exactly the reason cited: this is a situation where the FSF recommends the GPL over the LGPL. Especially since LilyPond is a GNU project.

Personally, I prefer the approach taken by (for example) Riverbank with PyQt, which I've also seen recommended (somewhere) by the FSF. If you reimburse the authors with money to support development, they will grant a proprietary license for PyQt, otherwise you must reimburse the community at large by contributing your own software.

For the conceivable future, this is always going to be one of the most tricky aspects of free software!

**Comment by dvzrv** [comment]
Whatever is/ was decided here, please add the exact version of the GPL license to this repository, so it can be used by package maintainers and be referenced properly. Thanks!
